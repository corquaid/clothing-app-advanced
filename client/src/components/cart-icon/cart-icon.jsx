import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { toggleCartHidden } from '../../redux/cart/cart.actions';
import { selectCartItemsCount } from '../../redux/cart/cart.selectors'; 

import './cart-icon.scss';
import { ReactComponent as ShoppingIcon } from '../../assets/shopping-bag.svg';

const CartIcon = ({ toggleCartHidden, totalItems }) => (
  <div className='cart-icon' onClick={toggleCartHidden}>
    <ShoppingIcon className='shopping-icon' />
    <span className='item-count'>{totalItems}</span>
  </div>
);

const mapDispatchToProps = dispatch => ({
  toggleCartHidden: () => dispatch(toggleCartHidden())
})

// Now mapStateToProps with the cart accumulator number to display on the cart icon REFERENCING SELECTORS

const mapStateToProps = createStructuredSelector({
  totalItems: selectCartItemsCount // from cart selector file
})

export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);