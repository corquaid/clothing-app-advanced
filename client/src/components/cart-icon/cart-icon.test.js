import React from "react";
import { shallow } from "enzyme";
import { CartIcon } from "./cart-icon";

describe("CartIcon component", () => {
    let wrapper;
    let mockToggleCartHidden;

    beforeEach(() => {
        mockToggleCartHidden = jest.fn();

        const mockProps = {
            totalItems: 0,
            toggleCartHidden: mockToggleCartHidden,
        };

        wrapper = shallow(<CartIcon {...mockProps} />);
    });

    it("Should render the CartIcon component", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("Should call toggleCartHidden when icon is clicked", () => {
        wrapper.find(".cart-icon").simulate("click");
        expect(mockToggleCartHidden).toHaveBeenCalled();
    });

    it("Should render the totalItems count as the component text", () => {
        const itemCount = parseInt(wrapper.find(".item-count").text());
        expect(itemCount).toBe(0);
    });
});
