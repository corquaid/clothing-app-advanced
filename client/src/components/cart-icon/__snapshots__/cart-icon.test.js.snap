// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component Should render the CartIcon component 1`] = `ShallowWrapper {}`;
